### PR TITLE
> 4x performance improvement

### PR DESCRIPTION
**Pull request by taylorking**
Caching the routing code and cors code increased throughput from 2k / sec to 12k requests /second

### CODE REVIEW

File: api-gateway-config/conf.d/managed_endpoints.conf [hunk 1]
```diff
@@ -67,21 +68,25 @@ server {
         set $cors_methods '';
         
         access_by_lua_block {
-            local routing = require "routing"
-            routing.processCall()
-            local cors = require "cors"
-            ngx.var.cors, ngx.var.cors_methods = cors.processCall(ngx.var["tenant"], ngx.var["gatewayPath"])   
+            if ngx.var.rounting == nil then
```
Comment (alexsong93) [inline api-gateway-config/conf.d/managed_endpoints.conf]: there is a typo here `rounting` -> `routing`

File: api-gateway-config/conf.d/managed_endpoints.conf [hunk 2]
```diff
@@ -67,21 +68,25 @@ server {
         set $cors_methods '';
         
         access_by_lua_block {
-            local routing = require "routing"
-            routing.processCall()
-            local cors = require "cors"
-            ngx.var.cors, ngx.var.cors_methods = cors.processCall(ngx.var["tenant"], ngx.var["gatewayPath"])   
+            if ngx.var.routing == nil then
```
Comment (ddragosd) [inline api-gateway-config/conf.d/managed_endpoints.conf]: when lua_code_cache is turned `on` lua modules are not loaded from scratch; there is 1 lua vm per nginx worker process. If there's a difference in this I'm thinking that there might be other factors that can influence the performance of the import.
See more at https://github.com/openresty/lua-nginx-module#lua_code_cache

**Comment by taylorking** [comment]
Ah @ddragosd could it be because we are using access_by_lua_block instead of access_by_lua_file?

**Comment by ddragosd** [comment]
`access_by_lua_block` should be fine. I also wanted to share the following link: 
https://github.com/openresty/lua-nginx-module#data-sharing-within-an-nginx-worker

**Comment by ddragosd** [comment]
I've also created 2 issues with some performance improvements tips: #120 and #121 .

**Comment by mhamann** [comment]
@ddragosd do you have concerns about merging this in?

**Pull request closed without merging**